src/phoenix/server/api_key_validator.py

```diff
@@ -0,0 +1,65 @@
+from asyncio import create_task, sleep
+from typing import Any, Dict, cast
+
+import jwt
+from jwt import InvalidTokenError
+from sqlalchemy import and_, select
+
+from phoenix.auth import JwtPayload
+from phoenix.config import PHOENIX_SECRET
+from phoenix.db import models
+from phoenix.server.types import DaemonTask, DbSessionFactory
+
+
+class ApiKeyValidator(DaemonTask):
```

Comment:
Did you opt for the daemon rather than the TTL cache to batch calls for different API keys?

Reply:
the issue is with replicas: api-key is created by one replica, but spans are received by another